.github/workflows/build.yml.in

```diff
@@ -84,6 +84,25 @@ jobs:
       - name: check that all files are imported
         run: git diff --exit-code
 
+  check_workflows:
+    if: github.repository MAIN_OR_FORK 'leanprover-community/mathlib4'
+    name: check workflowsJOB_NAME
+    runs-on: ubuntu-latest
+    steps:
+      - name: cleanup
+        run: |
+          find . -name . -o -prune -exec rm -rf -- {} +
```

Comment:
This could probably do with an explanation; I'm guessing the answer is "on the hoskinson runners we can't assume a clean checkout"?

Reply:
Though it looks to me like `actions/checkout` already ensures this with `clean: true` (the default)

Reply:
I'm basically cargo culting here, the other actions do it so it seems like it's probably a good thing. But I'm in no position to write the comment as I didn't add the lines! Maybe we can do some git digging to find out why this was introduced

Reply:
Ah, I hadn't noticed that